src/api/stats/avax/getPangolinV2DualApys.js

```diff
@@ -0,0 +1,156 @@
+const BigNumber = require('bignumber.js');
+const { MultiCall } = require('eth-multicall');
+const { avaxWeb3: web3, multicallAddress } = require('../../../utils/web3');
+
+const MasterChef = require('../../../abis/avax/PangolinChef.json');
+const Rewarder = require('../../../abis/avax/PangolinRewarderViaMultiplier.json');
+const ERC20 = require('../../../abis/ERC20.json');
+const fetchPrice = require('../../../utils/fetchPrice');
+const pools = require('../../../data/avax/pangolinV2DualLpPools.json');
+const { BASE_HPY, AVAX_CHAIN_ID } = require('../../../constants');
+const { getTradingFeeApr } = require('../../../utils/getTradingFeeApr');
+import { getFarmWithTradingFeesApy } from '../../../utils/getFarmWithTradingFeesApy';
+const { pangolinClient } = require('../../../apollo/client');
+const { compound } = require('../../../utils/compound');
+
+const masterchef = '0x1f806f7C8dED893fd3caE279191ad7Aa3798E928';
+const oracleIdA = 'PNG';
+const oracleA = 'tokens';
+const DECIMALSA = '1e18';
+
+const secondsPerYear = 31536000;
+
+const liquidityProviderFee = 0.0025;
+const beefyPerformanceFee = 0.045;
+const shareAfterBeefyPerformanceFee = 1 - beefyPerformanceFee;
+
+const getPangolinV2DualApys = async () => {
+  let apys = {};
+  let apyBreakdowns = {};
+
+  const tokenPriceA = await fetchPrice({ oracle: oracleA, id: oracleIdA });
+  const { rewardPerSecond, totalAllocPoint } = await getMasterChefData();
+  const { balances, allocPoints, multipliers } = await getPoolsData(pools);
+
+  const pairAddresses = pools.map(pool => pool.address);
+  const tradingAprs = await getTradingFeeApr(pangolinClient, pairAddresses, liquidityProviderFee);
+
+  for (let i = 0; i < pools.length; i++) {
+    const pool = pools[i];
+
+    const lpPrice = await fetchPrice({ oracle: 'lps', id: pool.name });
+    const totalStakedInUsd = balances[i].times(lpPrice).dividedBy('1e18');
+
+    const relativeRewards = rewardPerSecond.times(allocPoints[i]).dividedBy(totalAllocPoint);
+    const yearlyRewards = relativeRewards.times(secondsPerYear);
+    const yearlyRewardsAInUsd = yearlyRewards.times(tokenPriceA).dividedBy(DECIMALSA);
+    var yearlyRewardsBInUsd = new BigNumber(0);
+    var yearlyRewardsCInUsd = new BigNumber(0);
+
+    if (multipliers[i][0] != null) {
+      const tokenPriceB = await fetchPrice({ oracle: pool.oracleB, id: pool.oracleIdB });
+      const tokenBMultiplier = multipliers[i].map(v => v.multiplier[0]);
+      const yearlyRewardsB = yearlyRewards.times(tokenBMultiplier).dividedBy(pool.decimalsB);
+      yearlyRewardsBInUsd = yearlyRewardsB.times(tokenPriceB).dividedBy(pool.decimalsB);
+    }
+    if (multipliers[i][1] != null) {
+      const tokenPriceB = await fetchPrice({ oracle: pool.oracleB, id: pool.oracleIdC });
```

Comment:
please rewrite this block more carefully, without B values

Reply:
The implementation for the C token was actually work in progress, I should have marked the PR as draft.

I finished the implementation now and both additional rewards are properly calculated now.
For the 'oracle' variable I reused oracleB, as it's 'tokens' anyway, so I suggest we don't introduce a redundant property 'oracleC' to the pangolinV2DualLpPools.json.